### PR TITLE
live: Add `__enter__` and `__exit__`.

### PR DESCRIPTION
**Pull request by daavoo**
Wait for #353

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -355,3 +359,29 @@ def test_make_summary_without_calling_log(tmp_dir):
...
+        # no `step`
+        "foo": 1.0
+    }
+    log_file = tmp_dir / live.plots_dir / Metric.subfolder / "foo.tsv"
+    assert not log_file.exists()
+
+
+def test_context_manager(tmp_dir):
+    with Live() as live:
+        live.summary["foo"] = 1.0
```
Comment (daavoo) [inline tests/test_main.py:380]: `__exit__` only calls `live.make_summary` and `post_to_studio` for now. 

After https://github.com/iterative/dvc/pull/8408 is available, the context manager should take care of generating the DVC experiment

Comment (daavoo) [reply tests/test_main.py:380]: @dberenbaum Should `__exit__` also call `make_report()`? I think so

Comment (dberenbaum) [reply tests/test_main.py:380]: Yup, I agree, it should call `make_report()`.

**Review by dberenbaum — APPROVED** [review]

File: src/dvclive/live.py [hunk 2]
```diff
@@ -257,3 +259,9 @@ def read_step(self):
     def read_latest(self):
         with open(self.metrics_file, encoding="utf-8") as fobj:
             return json.load(fobj)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.end()
```
Comment (dberenbaum) [inline src/dvclive/live.py:267]: Minor follow-up: let's either make `end()` public or get rid of it and move the logic here. What do you think @daavoo?

Comment (dberenbaum) [reply src/dvclive/live.py:267]: I slightly prefer getting rid of `end()` since I'm not sure it's worth adding to docs, but no strong preference

**Pull request merged**